sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java

```diff
@@ -957,15 +957,26 @@ public void sessionCreated(IoSession session) throws Exception {
 
         @Override
         public void sessionClosed(IoSession session) throws Exception {
-            TcpipClientChannel channel = (TcpipClientChannel) session.removeAttribute(TcpipClientChannel.class);
+            TcpipClientChannel channel = (TcpipClientChannel) session.removeAttribute(TcpipClientChannel.class);                      
             Throwable cause = (Throwable) session.removeAttribute(TcpipForwardingExceptionMarker.class);
             if (channel != null) {
                 if (debugEnabled) {
                     log.debug("sessionClosed({}) closing channel={} after {} messages - cause={}",
-                            session, channel, messagesCounter, (cause == null) ? null : cause.getClass().getSimpleName());
+                        session, channel, messagesCounter, (cause == null) ? null : cause.getClass().getSimpleName());
+                }
+              
+                if (cause != null) {
+                    channel.close(true);
+                } else {
+                    OpenFuture openFuture = channel.getOpenFuture();
+                    if (!openFuture.isDone()) {
+                        openFuture.addListener(f -> {
+                            channel.close(false);        
+                        });
+                    } else {
+                        channel.close(false);                            
+                    }                  
                 }
```

Comment:
This code is not rebased on latest version...

Reply:
I agree, I changed only DefaultSshFuture.java for this issue.